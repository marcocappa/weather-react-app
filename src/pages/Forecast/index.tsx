import React, { useState } from 'react';
import Layout from '../../components/Layout';
import FilterInput from '../../components/FilterInput';

import { CITIES } from '../../common/constants';
import ForecastCard from '../../components/ForecastCard';

const Forecast = (): JSX.Element => {
    const [minTemperature, setMinTemperature] = useState('');
    const [maxTemperature, setMaxTemperature] = useState('');

    const handleMinTemperature = (e: React.ChangeEvent<HTMLInputElement>) => {
        setMinTemperature(e?.target?.value);
    };

    const handleMaxTemperature = (e: React.ChangeEvent<HTMLInputElement>) => {
        setMaxTemperature(e?.target?.value);
    };

    return (
        <Layout title="Forecast">
            <FilterInput
                label="Filter by Min temperature"
                onChange={handleMinTemperature}
            />
            <FilterInput
                label="Filter by Max temperature"
                onChange={handleMaxTemperature}
            />
            {minTemperature && <p>Showing items lower than {minTemperature}</p>}
            {maxTemperature && (
                <p>Showing items higher than {maxTemperature}</p>
            )}

            <h2 className="subheading">Here you can see the forecast:</h2>

            <div className="cards-container">
                {CITIES.map((location) => (
                    <ForecastCard
                        min={Number(minTemperature) || undefined}
                        max={Number(maxTemperature) || undefined}
                        key={location.CITY}
                        location={location}
                    />
                ))}
            </div>
        </Layout>
    );
};

export default Forecast;
