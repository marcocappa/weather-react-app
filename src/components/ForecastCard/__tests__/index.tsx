import React from 'react';
import { shallow } from 'enzyme';
import Forecast from '../index';

describe('<Forecast />', () => {
    const props = {
        location: {
            CITY: 'London',
            COUNTRY: 'UK',
        },
        min: undefined,
        max: undefined,
    };

    it('should match snapshot', () => {
        const wrapper = shallow(<Forecast {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
