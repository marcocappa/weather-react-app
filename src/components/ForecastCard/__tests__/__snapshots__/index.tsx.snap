// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forecast /> should match snapshot 1`] = `
<Fragment>
  <div
    className="forecast-card"
  >
    <p>
      loading...
    </p>
  </div>
</Fragment>
`;
